src/v/config/property.h

```diff
@@ -356,6 +356,8 @@ concept is_collection = requires(T x) {
     typename T::value_type;
     !std::is_same_v<typename T::value_type, char>;
     {x.size()};
+    {x.begin()};
```

Comment:
Should you check that it return iterator type?

Reply:
presumably at some point maybe after we upgrade clang we can do `constexpr if (std::Container<T>)`

Reply:
> Should you check that it return iterator type?

it just depends on how strict we want to be. something with these 4 properties are likely to do what we want.

Reply:
I was kind of sneaking this slight improvement to the concept in as a nice to have, making it even more robust is probably a task for another day.